client_token_integration_test.go

```diff
@@ -8,39 +8,69 @@ import (
 )
 
 // This test will fail unless you set up your Braintree sandbox account correctly. See TESTING.md for details.
-func TestClientToken(t *testing.T) {
-	t.Parallel()
-
-	ctx := context.Background()
-
-	g := testGateway.ClientToken()
-	token, err := g.Generate(ctx)
-	if err != nil {
-		t.Fatalf("failed to generate client token: %s", err)
-	}
-	if len(token) == 0 {
-		t.Fatalf("empty client token!")
-	}
-}
-
-func TestClientTokenWithCustomer(t *testing.T) {
-	t.Parallel()
-
-	ctx := context.Background()
-
-	customerRequest := &CustomerRequest{FirstName: "Lionel"}
-
-	customer, err := testGateway.Customer().Create(ctx, customerRequest)
+func TestClientTokenGateway_Generate(t *testing.T) {
+	// Getting customer from the API
+	customer, err := testGateway.Customer().Create(context.Background(), &CustomerRequest{FirstName: "Lionel"})
 	if err != nil {
 		t.Error(err)
 	}
 
-	customerId := customer.Id
-
-	token, err := testGateway.ClientToken().GenerateWithCustomer(ctx, customerId)
-	if err != nil {
-		t.Error(err)
-	} else if len(token) == 0 {
-		t.Fatalf("Received empty client token")
+	tests := []struct {
+		name      string
+		req       *ClientTokenRequest
+		wantError bool
+	}{
+		{
+			name: "empty request struct",
+			req:  nil,
+		},
+		{
+			name: "request with provided version",
+			req:  &ClientTokenRequest{Version: 2},
+		},
+		{
+			name:      "request with non existent customerID",
+			req:       &ClientTokenRequest{CustomerID: "///////@@@@@@@"},
+			wantError: true,
+		},
+		{
+			name: "request with customer id",
+			req:  &ClientTokenRequest{CustomerID: customer.Id},
+		},
+		{
+			name: "request with merchant id",
+			req:  &ClientTokenRequest{MerchantAccountID: testMerchantAccountId},
+		},
+		{
+			name: "request with customer id and merchant id",
+			req:  &ClientTokenRequest{CustomerID: customer.Id, MerchantAccountID: testMerchantAccountId},
+		},
+		{
+			name: "request with customer id and merchant id and options",
+			req: &ClientTokenRequest{
+				CustomerID:        customer.Id,
+				MerchantAccountID: testMerchantAccountId,
+				Options: &ClientTokenRequestOptions{
+					FailOnDuplicatePaymentMethod: true,
+					MakeDefault:                  true,
+					VerifyCard:                   true,
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			g := testGateway.ClientToken()
+			token, err := g.Generate(context.TODO(), tt.req)
+			if (err != nil) != tt.wantError {
+				t.Errorf("ClientTokenGateway.Generate() error = %v, wantErr = %v", err, tt.wantError)
+				return
+			}
+			if !tt.wantErr && len(token) == 0 {
```

Comment:
This line has a typo on it that is what's causing the CI builds to fail. `wantError` is the name of the field on `tt`.

Aside to that though, I think `err` is a common name for an identifier containing an error, so think it's worth renaming the field to `wantErr` to match this instead.